libs/feature/map/src/lib/add-layer-from-file/add-layer-from-file.component.ts

```diff
@@ -0,0 +1,110 @@
+import { ChangeDetectorRef, Component } from '@angular/core'
+import { MapContextLayerModel } from '../map-context/map-context.model'
+import { MapFacade } from '../+state/map.facade'
+
+const INVALID_FILE_FORMAT_ERROR_MESSAGE = 'Invalid file format'
+
+@Component({
+  selector: 'gn-ui-add-layer-from-file',
+  templateUrl: './add-layer-from-file.component.html',
+  styleUrls: ['./add-layer-from-file.component.css'],
+})
+export class AddLayerFromFileComponent {
+  errorMessage: string | null = null
+  successMessage: string | null = null
+  loading = false
+  readonly acceptedMimeType = ['.geojson']
+  readonly maxFileSize = 5000000
+
+  constructor(
+    private mapFacade: MapFacade,
+    private changeDetectorRef: ChangeDetectorRef
+  ) {}
+
+  async handleFileChange(file: File) {
+    if (!file) {
+      this.displayMessage(INVALID_FILE_FORMAT_ERROR_MESSAGE, 'error')
+      return
+    }
+    if (file.size > this.maxFileSize) {
+      this.displayMessage('File size exceeds the limit of 5MB', 'error')
+      return
+    }
+    await this.addLayer(file)
+  }
+
+  private async addLayer(file: File) {
+    this.errorMessage = null
+    this.loading = true
+    try {
+      if (!this.isFileFormatValid(file)) {
+        this.displayMessage(INVALID_FILE_FORMAT_ERROR_MESSAGE, 'error')
+        return
+      }
+
+      const fileExtension = this.getFileExtension(file)
+      switch (fileExtension) {
```

Comment:
One improvement here would be to use DuckDB  spatial WASM, to be able to load any kind of spatial dataset.